src/viam/ur/module/ur_arm.cpp

```diff
@@ -735,16 +735,49 @@ void URArm::move_joint_space_(std::shared_lock<std::shared_mutex> config_rlock,
         return;
     }
 
+    // Walk all interior points of the waypoints list, if any. If the
+    // point of current interest is the cusp of a direction reversal
+    // w.r.t. the points immediately before and after it, then splice
+    // all waypoints up to and including the cusp point into a new
+    // segment, and then begin accumulating a new segment starting at
+    // the cusp point. The cusp point is duplicated, forming both the
+    // end of one segment and the beginning of the next segment. After
+    // exiting the loop, any remaining waypoints form the last (and if
+    // no cusps were identified the only) segment. If one or more cusp
+    // points were identified, the waypoints list will always have at
+    // least two residual waypoints, since the last waypoint is never
+    // examined, and the splice call never removes the waypoint being
+    // visited.
+    //
+    // NOTE: This assumes waypoints have been de-duplicated to avoid
+    // zero-length segments that would cause numerical issues in
+    // normalized() calculations.
+    std::vector<decltype(waypoints)> segments;
+    for (auto where = next(begin(waypoints)); where != prev(end(waypoints)); ++where) {
+        const auto segment_ab = *where - *prev(where);
+        const auto segment_bc = *next(where) - *where;
+        const auto dot = segment_ab.normalized().dot(segment_bc.normalized());
+        if (std::fabs(dot + 1.0) < 1e-3) {
+            segments.emplace_back();
+            segments.back().splice(segments.back().begin(), waypoints, waypoints.begin(), where);
+            segments.back().push_back(*where);
+        }
+    }
+    segments.push_back(std::move(waypoints));
+
     const auto new_trajectory = [&]() -> std::optional<std::vector<trajectory_sample_point>> {
         if (!current_state_->use_new_trajectory_planner()) {
             return std::nullopt;
         }
 
+        size_t total_waypoints = 0;
+        double total_duration = 0.0;
+        viam::trajex::arc_length total_arc_length{0.0};
+
         try {
             using namespace viam::trajex;
 
-            const auto waypoints_xarray = eigen_waypoints_to_xarray(waypoints);
-            const totg::waypoint_accumulator trajex_waypoints(waypoints_xarray);
+            const auto generation_start = std::chrono::steady_clock::now();
```

Comment:
nit: before, we weren't measuring the time to build the max_velocity and mac_acceleration vectors. I mean, it's nanoseconds, so who cares, but it something we don't care to measure. In the future, when we are fully on trajex, these will be pre-computed and we won't need to make them on the fly.